tests/unit/frameworks/kfp/constructs/scripts_test.py

```diff
@@ -0,0 +1,476 @@
+            f'\n'
+            f'  echo "Creating Cloudbuild Trigger on branch $CLOUD_SOURCE_REPO_BRANCH in project $PROJECT_ID for repo ${LEFT_BRACKET}CLOUD_SOURCE_REPO{RIGHT_BRACKET}"\n'
+            f'  gcloud beta builds triggers create cloud-source-repositories \{NEWLINE}'
+            f'  --region=$CB_TRIGGER_LOCATION \{NEWLINE}'
+            f'  --name=$CB_TRIGGER_NAME \{NEWLINE}'
+            f'  --repo=$CLOUD_SOURCE_REPO \{NEWLINE}'
+            f'  --branch-pattern="$CLOUD_SOURCE_REPO_BRANCH" \{NEWLINE}'
+            f'  --build-config=base_dircloudbuild.yaml\n'
+            f'\n'
+            f'else\n'
+            f'\n'
+            f'  echo "Cloudbuild Trigger already exists in project $PROJECT_ID for repo ${LEFT_BRACKET}CLOUD_SOURCE_REPO{RIGHT_BRACKET}"\n'
+            f'\n'
+            f'fi\n')
+
+        assert scripts.dockerfile == (
+            GENERATED_LICENSE +
+            f'FROM {base_image}\n'
+            f'RUN python -m pip install --upgrade pip\n'
+            f'COPY requirements.txt .\n'
+            f'RUN python -m pip install -r \ \n'
+            f'    requirements.txt --quiet --no-cache-dir \ \n'
+            f'    && rm -f requirements.txt\n'
+            f'COPY ./src /pipelines/component/src\n'
+            f'ENTRYPOINT ["/bin/bash"]\n')
+
+        assert scripts.defaults == (
+            GENERATED_LICENSE +
+            f'# These values are descriptive only - do not change.\n'
+            f'# Rerun AutoMLOps.generate() to change these values.\n'
+            f'gcp:\n'
+            f'  af_registry_location: {af_registry_location}\n'
+            f'  af_registry_name: {af_registry_name}\n'
+            f'  base_image: {base_image}\n'
+            f'  cb_trigger_location: {cb_trigger_location}\n'
+            f'  cb_trigger_name: {cb_trigger_name}\n'
+            f'  cloud_run_location: {cloud_run_location}\n'
+            f'  cloud_run_name: {cloud_run_name}\n'
+            f'  cloud_tasks_queue_location: {cloud_tasks_queue_location}\n'
+            f'  cloud_tasks_queue_name: {cloud_tasks_queue_name}\n'
+            f'  cloud_schedule_location: {schedule_location}\n'
+            f'  cloud_schedule_name: {schedule_name}\n'
+            f'  cloud_schedule_pattern: {schedule_pattern}\n'
+            f'  cloud_source_repository: {csr_name}\n'
+            f'  cloud_source_repository_branch: {csr_branch_name}\n'
+            f'  gs_bucket_name: {gs_bucket_name}\n'
+            f'  pipeline_runner_service_account: {pipeline_runner_sa}\n'
+            f'  project_id: {project_id}\n'
+            f'  vpc_connector: {vpc_connector}\n'
+            f'\n'
+            f'pipelines:\n'
+            f'  parameter_values_path: {tmpdir}\n'
+            f'  pipeline_component_directory: components\n'
+            f'  pipeline_job_spec_path: {tmpdir}\n'
+            f'  pipeline_region: {gs_bucket_location}\n'
+            f'  pipeline_storage_path: gs://{gs_bucket_name}/pipeline_root\n')
+
+        default_reqs = [
```

Comment:
This tests the scenario where the user inputs reqs. Are we testing the package dependency inferring using pipreqs? This might be something that should be it's own test, it would be good to have coverage over it though

Reply:
Are you saying that when a user inputs their reqs, the scripts.requirements attribute should hold not only what the user inputted but the inferred packages as well? I'm not too sure how to approach this as the test currently takes reqs fed in as a list during parameterization. I believe the test call still runs through the whole create_requirements method in scripts.py including those inference steps.

Reply:
We'll address the pipreqs testing under integration test suite, rather than unit tests